zerver/tests/test_subs.py

```diff
@@ -2771,6 +2809,10 @@ def test_basic_add_delete(self) -> None:
         streams = self.get_streams(user)
         self.assertTrue("my_test_stream_1" in streams)
 
+        stream = get_stream("my_test_stream_1", user.realm)
```

Comment:
This seemed to be a little confusing. If a user creates a stream (and becomes the stream admin for it in the process), then doesn't their role entitle them to delete the stream? Converting the admin to a member, before deleting the stream seems a little odd.

Reply:
Actually the url used in this test is not for deleting/deactivating the stream, but instead of unsubscribing the user, which in turn deletes the stream as there are no subscribers. So I am changing the role to member, as we do not allow last stream admin to unsubscribe. Maybe the confusion was there due to comment in that test, we can correct that comments.

Reply:
On second thought, we can remove the restriction of removing of not allowing the last admin to unsubscribe if the stream admin is only subscriber in the stream. What are your thoughts on this?

Reply:
Yup exactly. We should just allow the last member (who would be an admin as per our previous conditions) to delete the stream (i.e unsubscribe) without changing their role to a member. Want to bring this up on czo? :)

Reply:
Started the converstaion https://chat.zulip.org/#narrow/stream/3-backend/topic/stream.20administrators/near/1122637 on czo.